trunk/source/Library-UltimateModel/src/de/uni_freiburg/informatik/ultimate/core/model/models/annotation/ISpec.java

```diff
@@ -0,0 +1,332 @@
+/*
+ * Copyright (C) 2023 Manuel Bentele
+ *
+ * This file is part of the ULTIMATE Core.
+ *
+ * The ULTIMATE Core is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ULTIMATE Core is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with the ULTIMATE Core. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Additional permission under GNU GPL version 3 section 7:
+ * If you modify the ULTIMATE Core, or any covered work, by linking
+ * or combining it with Eclipse RCP (or a modified version of Eclipse RCP),
+ * containing parts covered by the terms of the Eclipse Public License, the
+ * licensors of the ULTIMATE Core grant you additional permission
+ * to convey the resulting work.
+ */
+
+package de.uni_freiburg.informatik.ultimate.core.model.models.annotation;
+
+import java.util.Set;
+
+/**
+ * Defines specification types for arbitrary checks and analyzes.
+ * 
+ * @author Manuel Bentele
+ */
+public interface ISpec {
```

Comment:
I am not quite sure that the new architecture for specification types is an improvement above the old one. Could you explain to me why it is better like this?

Reply:
I quickly moved the `Spec` types to an interface because they have no place in the implementation. In my opinion these are part of the generic software model as they can be implemented anywhere in Ultimate (see program checks in `Check` as well as requirement checks in `ReqCheck`).

However, we don't necessarily need the interface. On the one hand, it would be currently sufficient if only the `ISpec.Type` enum is made public available in the model. On the other hand, additional functions could be added to the interface if the `Check` class is cleaned up. Currently, the connections between `ReqCheck`, `Check` and `Spec`s looked a bit hacky to me and were not properly encapsulated.

@schuessf also wanted the addition of new `ISpec.Type`s to be easier and the default messages to be defined centrally. Instead of having to edit three files, you now have to edit only one file.

Do you have another idea for the general software structure regarding `Check`s and `Spec` types?